src/ml/algorithm_analysis.py

```diff
@@ -0,0 +1,131 @@
+import pandas as pd
+from typing import Iterable
+from pathlib import Path
+import os
+
+import matplotlib.pyplot as plt
+from sklearn.decomposition import PCA
+from sklearn.preprocessing import StandardScaler
+import seaborn as sns
+
+from sklearn.cluster import AgglomerativeClustering
+from scipy.cluster.hierarchy import dendrogram
+import numpy as np
+
+import argparse
+
+def is_file_empty(filepath):
+    return os.path.getsize(filepath)==0
+
+def summarize_networks(file_paths: Iterable[Path]) -> pd.DataFrame:
+
+    edge_tuples = []
+    
+    for file in file_paths:
+        
+        try:
+           
+            with open(file,'r') as f:
+                lines = [line[:3] for line in f.readlines()]
+            
+            line = []
+            for char in lines:
+                newChar = char.replace(' ','')
+                line.append(newChar)
+                
+            e = [''.join(sorted(ele)) for ele in line]
+
+            # for smaller labels (havent put it yet because of the tester files, but will be changed to this)
+            parts = file.split('/')
+            edge_tuples.append((parts[-2], e))
+  
+        except FileNotFoundError:
+            print(file, ' DOES NOT EXIST') # should hopefully not hit
+            continue
+        
+    edge_dataframes = []
+    
+    for tuple in edge_tuples:
+        col_label = str(tuple[0])
+
+        dataframe = pd.DataFrame(
+            {
+                col_label: 1,
+            }, index= tuple[1]
+        )
+        edge_dataframes.append(dataframe)
+
+    concated_df = pd.concat(edge_dataframes, axis= 1, join = 'outer')
+    concated_df = concated_df.fillna(0) 
+    concated_df = concated_df.astype('int64')
+   
+    return concated_df
+    
+def pca(dataframe: pd.DataFrame, output_png: str, output_file: str):
+
+    df = dataframe.reset_index(drop=True)
+    df = df.transpose() #based on the algortihms rather than the edges 
+    X = df.values
+
+    # standardize the df because if not in the same scale (rn it is 4,7)
```

Comment:
What is the (4,7) comment referring to?

Reply:
old problem, was a problem prior to transposing the dataframe